pip/req/req_install.py

```diff
@@ -361,7 +361,7 @@ def _correct_build_location(self):
     def name(self):
         if self.req is None:
             return None
-        return native_str(self.req.project_name)
+        return native_str(pkg_resources.safe_name(self.req.name))
```

Comment:
Ah, but canonicalize_name also lowercases the result, which is bad, since it turns up in the output of pip, which breaks the test suite, and also impacts uninstallability.